src/NuGet.Core/NuGet.Configuration/Properties/AssemblyInfo.cs

```diff
@@ -11,4 +11,4 @@
 [assembly: InternalsVisibleTo("NuGet.VisualStudio.Implementation.Test, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
 [assembly: InternalsVisibleTo("NuGet.Configuration.Test, PublicKey=0024000004800000940000000602000000240000525341310004000001000100b5fc90e7027f67871e773a8fde8938c81dd402ba65b9201d60593e96c492651e889cc13f1415ebb53fac1131ae0bd333c5ee6021672d9718ea31a8aebd0da0072f25d87dba6fc90ffd598ed4da35e44c398c454307e8e33b8426143daec9f596836f97c8f74750e5975c64e2189f45def46b2a2b1247adc3652bf5c308055da9")]
 [assembly: InternalsVisibleTo("DynamicProxyGenAssembly2, PublicKey=0024000004800000940000000602000000240000525341310004000001000100c547cac37abd99c8db225ef2f6c8a3602f3b3606cc9891605d02baa56104f4cfc0734aa39b93bf7852f7d9266654753cc297e7d2edfe0bac1cdcf9f717241550e0a7b191195b7667bb4f64bcb8e2121380fd1d9d46ad2d92d2d15605093924cceaf74c4861eff62abf69b9291ed0a340e113be11e6a7d3113e92484cf7045cc7")]
-
+[assembly: InternalsVisibleTo("Test.Utility, PublicKey=002400000480000094000000060200000024000052534131000400000100010007d1fa57c4aed9f0a32e84aa0faefd0de9e8fd6aec8f87fb03766c834c99921eb23be79ad9d5dcc1dd9ad236132102900b723cf980957fc4e177108fc607774f29e8320e92ea05ece4e821c0a5efe8f1645c4c0c93c1ab99285d622caa652c1dfad63d745d6f2de5f17e5eaf0fc4963d261c8a12436518206dc093344d5ad293")]
```

Comment:
Why is this needed?  What internal API from `NuGet.Configuration` is `Test.Utility` using?

Reply:
@zivkan 
It's used here.
https://github.com/NuGet/NuGet.Client/blob/cd77b927c2e12d315113898ba8dc05801af48d60/test/TestUtilities/Test.Utility/PackageNamespacesConfigurationUtility.cs#L35

Here is actual api:
https://github.com/NuGet/NuGet.Client/blob/9a72664c07aaef63e5d23a529e34f292f96e99d7/src/NuGet.Core/NuGet.Configuration/PackageNamespaces/PackageNamespacesConfiguration.cs#L41-L46

Reply:
Wouldn't customers using our NuGet.Configuration package directly benefit from being able to use this API to mock configurations in their own tests, or even have hard-coded namespace configuration in their production applications, without having to read a file off disk?

Why isn't that API public?

Reply:
I think we should err on the side of underexposing than overexposing APIs tbh. 

Given that modes are coming at a later point, it might be better to consider that later.